### PR TITLE
Revert "Revert "Review model""

### PR DESCRIPTION
**Pull request by Jacopobelli5**
Made a mess. I merged by mistake, then i reverted and now i am pulling again by reverting this one. to see my comments go on pull number 22. sorry

### CODE REVIEW

File: app/models/boat.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ class Boat < ApplicationRecord
   # associations
   belongs_to :user
   has_many :bookings, dependent: :destroy
+  has_many :reviews
```
Comment (sarhan-a) [inline app/models/boat.rb:5]: has_many :reviews, dependent: :destroy

Comment (sarhan-a) [reply app/models/boat.rb:5]: If we delete a boat we should also delete it's reviews, I think

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class ReviewsController < ApplicationController
+  before_action :set_boat, only: [ :new, :create ]
+
+  def new
+    @review = Review.new
```
Comment (sarhan-a) [inline app/controllers/reviews_controller.rb:5]: Shouldn't we make a reference to the boat that is reviewed here? Remember we had an error the other day with the Booking controller at this level.

**Review by sarhan-a — CHANGES_REQUESTED** [review]

**Comment by sarhan-a** [comment]
Looks good, made some comments and a suggestion. Have a look please.

File: app/controllers/reviews_controller.rb (hunk 2 above)
Comment (Jacopobelli5) [reply app/controllers/reviews_controller.rb:5]: i know but it works this way i am telling you

File: app/models/boat.rb (hunk 1 above)
Comment (Jacopobelli5) [reply app/models/boat.rb:5]: i know i forgot that, we can add that later..

**Pull request merged**